File: CS/GridHorizontalScrolling/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace GridHorizontalScrolling {
    public partial class Form1 : Form {
        BindingList<Customers> listOfCustomers = new BindingList<Customers>();
        GridEcxtensionHorizontalScrolling Extension;

        public Form1() {
            InitializeComponent();
            gridControl1.DataSource = FillTables();
            gridView1.Columns[0].Fixed = DevExpress.XtraGrid.Columns.FixedStyle.Left;
            gridView1.LeftCoordChanged += gridView1_LeftCoordChanged;
            Extension = new GridEcxtensionHorizontalScrolling(gridView1);
        }

        void gridView1_LeftCoordChanged(object sender, EventArgs e) {
            textEdit1.Text = Extension.UpdateIndexInEditor().ToString();
        }

        BindingList<Customers> FillTables() {
            listOfCustomers.Clear();
            for (int i = 0; i < 5; i++)
                listOfCustomers.Add(new Customers());
            return listOfCustomers;
        }

        private void Form1_Load(object sender, EventArgs e) {
            Extension.UpdateFixedColumnsCount();
            textEdit1.Text = Extension.UpdateIndexInEditor().ToString();
        }

        private void simpleButton1_Click(object sender, EventArgs e) {
            Extension.ScrollForward();
        }

        private void simpleButton2_Click(object sender, EventArgs e) {
            Extension.ScrollBackward();
        }

        private void textEdit1_KeyDown(object sender, KeyEventArgs e) {
            if (e.KeyCode == Keys.Enter)
                if (textEdit1.Text != string.Empty)
                    Extension.ScrollTo(Convert.ToInt32(textEdit1.Text));
        }
    }
}

File: VB/GridHorizontalScrolling/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms

Namespace GridHorizontalScrolling
	Partial Public Class Form1
		Inherits Form
		Private listOfCustomers As New BindingList(Of Customers)()
		Private Extension As GridEcxtensionHorizontalScrolling

		Public Sub New()
			InitializeComponent()
			gridControl1.DataSource = FillTables()
			gridView1.Columns(0).Fixed = DevExpress.XtraGrid.Columns.FixedStyle.Left
			AddHandler gridView1.LeftCoordChanged, AddressOf gridView1_LeftCoordChanged
			Extension = New GridEcxtensionHorizontalScrolling(gridView1)
		End Sub

		Private Sub gridView1_LeftCoordChanged(ByVal sender As Object, ByVal e As EventArgs)
			textEdit1.Text = Extension.UpdateIndexInEditor().ToString()
		End Sub

		Private Function FillTables() As BindingList(Of Customers)
			listOfCustomers.Clear()
			For i As Integer = 0 To 4
				listOfCustomers.Add(New Customers())
			Next i
			Return listOfCustomers
		End Function

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			Extension.UpdateFixedColumnsCount()
			textEdit1.Text = Extension.UpdateIndexInEditor().ToString()
		End Sub

		Private Sub simpleButton1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles simpleButton1.Click
			Extension.ScrollForward()
		End Sub

		Private Sub simpleButton2_Click(ByVal sender As Object, ByVal e As EventArgs) Handles simpleButton2.Click
			Extension.ScrollBackward()
		End Sub

		Private Sub textEdit1_KeyDown(ByVal sender As Object, ByVal e As KeyEventArgs) Handles textEdit1.KeyDown
			If e.KeyCode = Keys.Enter Then
				If textEdit1.Text <> String.Empty Then
					Extension.ScrollTo(Convert.ToInt32(textEdit1.Text))
				End If
			End If
		End Sub
	End Class
End Namespace